compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.h

```diff
@@ -8,9 +8,12 @@
 #define IREE_COMPILER_CODEGEN_LLVMGPU_KERNELCONFIG_H_
 
 #include "mlir/IR/BuiltinOps.h"
+#include "mlir/Interfaces/FunctionInterfaces.h"
 
 namespace mlir::iree_compiler {
 
+int64_t getTargetSharedMemoryLimit(FunctionOpInterface entryPoint);
+
```

Comment:
I'm not sure if this is the right layering. Would be good if someone can confirm.

Reply:
Ideally I think we should have an attribute for it and put it in `IREECodegenAttrs.h`, or pull all of these target query function out of `KernelConfig.cpp`.

Reply:
This might need a bigger refactor than this patch could hold. I'm going to leave this function here for now and add a TODO.